hubclient/client.go

```diff
@@ -335,6 +417,48 @@ func (c *Client) HttpPostJSONExpectResult(url string, data interface{}, result i
 	return resp.Header.Get("Location"), nil
 }
 
+func (c *Client) HttpPostFile(url string, filePath string, contentType string) (string, int, error) {
+
+	var resp *http.Response
+	var err error
+
+	if c.debugFlags&HubClientDebugTimings != 0 {
+		log.Debugf("DEBUG HTTP STARTING POST REQUEST: %s", url)
+	}
+
+	file, err := os.Open(filePath)
+
+	if err != nil {
+		log.Errorf("Error opening file: %+v", err)
+		return "", -1, err
+	}
+
+	httpStart := time.Now()
+	req, err := http.NewRequest(http.MethodPost, url, file)
+	req.Header.Set(HeaderNameContentType, contentType)
+
+	if err != nil {
+		log.Errorf("Error making http post request: %+v.", err)
+		return "", -1, err
+	}
+
+	c.doPreRequest(req)
+	log.Debugf("POST Request: %+v.", req)
```

Comment:
If this call will log request's headers, they will contain basic auth header, and will be visible in a log file too